pyfolio/perf_attrib.py

```diff
@@ -612,6 +612,8 @@ def _align_and_warn(returns,
             )
             warnings.warn(warning_msg)
 
+    factor_loadings.index = factor_loadings.index.set_names(['dt', 'ticker'])
```

Comment:
I'm very nervous about mutating the input frame in place like this. Why do we depend specifically on the names of the index levels? If having index level names is absolutely required for this function to work correctly, then IMO we should treat it as an error if they're not what we expect.

Reply:
We want the indexes to align with positions - this lets us pass the output of `get_factor_loadings` to perf attrib directly

Reply:
I understand that we use the indexes for alignment, but does `perf_attrib` need to depend on the names of the levels of the indexes? As a consumer of this API, I would be surprised (and angry) if the dataframe I passed this function had the level names of its index silently changed.

Reply:
Ah makes sense. Updated this to not change the original input's index names